torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -1310,27 +1313,53 @@ TensorViewBuilder& TensorViewBuilder::shape(std::vector<Val*> shape) {
   return *this;
 }
 
+TensorViewBuilder& TensorViewBuilder::expanded(std::vector<bool> expanded) {
+  TORCH_CHECK(expanded_.empty(), "Attempting to reset expanded shape");
+  if (!expanded.empty()) {
+    TORCH_CHECK(ndims_ == 0 || ndims_ == expanded.size());
+    ndims_ = expanded.size();
+  }
+  expanded_ = std::move(expanded);
+  return *this;
+}
+
 TensorView* TensorViewBuilder::build() const {
   // Build the domain
   std::vector<IterDomain*> domain(ndims_, nullptr);
   for (const auto i : c10::irange(ndims_)) {
+    bool is_expanded = false;
+    Val* extent = nullptr;
+    Val* expanded_extent = nullptr;
+    Val** shape_extent = &extent;
```

Comment:
What is `shape_extent`? Why is this necessary?

Reply:
added a comment above describing what it does